packages/db/src/rules.ts

```diff
@@ -0,0 +1,62 @@
+import Knex from 'knex';
+import { RuleRecord } from '@cumulus/types/api/rules';
+import { PostgresRule, PostgresCollectionRecord, PostgresProviderRecord } from './types';
+import { getRecordCumulusId } from './database';
+import { tableNames } from './tables';
+
+const getCollectionCumulusId = async (record: RuleRecord, dbClient: Knex) => {
+  if (record.collection) {
+    await getRecordCumulusId<PostgresCollectionRecord>(
+      { name: record.collection.name, version: record.collection.version },
+      tableNames.collections,
+      dbClient
+    );
+  }
+  return undefined;
+};
+
+const getProviderCumulusId = async (record: RuleRecord, dbClient: Knex) => {
+  if (record.provider) {
+    const result = await getRecordCumulusId<PostgresProviderRecord>(
+      { name: record.provider },
+      tableNames.providers,
+      dbClient
+    );
+    return result;
+  }
+  return undefined;
+};
+
+/**
+ * Generate a Postgres rule record from a DynamoDB record.
+ * @param {Object} record - A rule
+ * @param {Object} dbClient - Knex client for reading from RDS database
+ * @returns {Object} A rule record
+ */
+export const translateApiRuleToPostgresRule = async (
+  record: RuleRecord,
+  dbClient: Knex
+): Promise<PostgresRule> => {
+  const collectionCumulusId = await getCollectionCumulusId(record, dbClient);
+  const providerCumulusId = await getProviderCumulusId(record, dbClient);
+  const ruleRecord: PostgresRule = {
+    name: record.name,
+    workflow: record.workflow,
+    provider_cumulus_id: record.provider ? providerCumulusId : undefined,
+    collection_cumulus_id: record.collection ? collectionCumulusId : undefined,
+    meta: (record.meta ? JSON.stringify(record.meta) : undefined),
```

Comment:
Ooh, noted.

Reply:
@markdboyd Taking a closer look at this, record.meta looks like:

```
  meta?: {
    retries?: number,
    visibility?: number,
    [key: string]: unknown
  },
```
but meta expects 
```
export interface PostgresRule {
  name: string,
  workflow: string,
  type: string,
  enabled: boolean,
  collection_cumulus_id?: number,
  provider_cumulus_id?: number,
  execution_name_prefix?: string,
  value?: string,
  arn?: string,
  log_event_arn?: string,
  payload?: object,
  meta?: string,
  tags?: string,
  queue_url?: string,
  created_at: Date,
  updated_at: Date,
}
```
Should I have the Postgres `meta` match the object definition I shared?

Reply:
`meta` should definitely at least be an `object` type. But in reality there's nothing in the postgres database schema that will enforce it to match a specific structure - it would accept any valid JSON. So I think `object` is fine